model/project_ref.go

```diff
@@ -2309,24 +2308,8 @@ func (p *ProjectRef) GetActivationTimeForVariant(variant *BuildVariant) (time.Ti
 
 // GetActivationTimeForTask returns the time at which this task should next be activated.
 // Temporarily takes in the task ID that prompted this query, for logging.
-func (p *ProjectRef) GetActivationTimeForTask(t *BuildVariantTaskUnit, taskId string) (time.Time, error) {
+func (p *ProjectRef) GetActivationTimeForTask(t *BuildVariantTaskUnit) (time.Time, error) {
 	defaultRes := time.Now()
-	// Verify that we mean to be getting activation time for task
-	if !t.HasSpecificActivation() {
```

Comment:
Moved this out of the function and into the generate tasks usage case to avoid adding unnecessary build variant task statuses to the version.